test/fn_api_test.c

```diff
@@ -163,35 +217,43 @@ static struct test_case_st test_add_cases[] = {
 
 static int test_add(int i)
 {
-    const OSSL_FN_ULONG *n1 = test_add_cases[i].op1;
-    size_t n1_limbs = test_add_cases[i].op1_size;
-    const OSSL_FN_ULONG *n2 = test_add_cases[i].op2;
-    size_t n2_limbs = test_add_cases[i].op2_size;
-    const OSSL_FN_ULONG *ex = test_add_cases[i].ex;
-    size_t ex_limbs = test_add_cases[i].ex_size;
-    size_t check_limbs = test_add_cases[i].check_size;
-    int ret = 1;
-    OSSL_FN *fn1 = NULL, *fn2 = NULL, *res = NULL;
-    const OSSL_FN_ULONG *u = NULL;
-
-    /* To test that OSSL_FN_add() does a complete job, 'res' is pre-polluted */
+    return test_add_common(test_add_cases[i]);
+}
 
-    if (!TEST_ptr(fn1 = OSSL_FN_new_limbs(n1_limbs))
-        || !TEST_ptr(fn2 = OSSL_FN_new_limbs(n2_limbs))
-        || !TEST_ptr(res = OSSL_FN_new_limbs(ex_limbs))
-        || !TEST_true(pollute(res, 0, ex_limbs))
-        || !TEST_true(ossl_fn_set_words(fn1, n1, n1_limbs))
-        || !TEST_true(ossl_fn_set_words(fn2, n2, n2_limbs))
-        || !TEST_true(OSSL_FN_add(res, fn1, fn2))
-        || !TEST_ptr(u = ossl_fn_get_words(res))
-        || !TEST_mem_eq(u, ossl_fn_get_dsize(res) * OSSL_FN_BYTES,
-                        ex, check_limbs * OSSL_FN_BYTES))
-        ret = 0;
-    OSSL_FN_free(fn1);
-    OSSL_FN_free(fn2);
-    OSSL_FN_free(res);
+#define ADD_TRUNCATED_CASE(i, op1, op2, ex, exp)        \
+    {                                                   \
+        /* op1, with size */ op1, LIMBSOF(op1),         \
+        /* op2, with size */ op2, LIMBSOF(op2),         \
+        /* ex, with size */ ex, LIMBSOF(ex),            \
+        /* op1_live_size */ LIMBSOF(op1) + 1,           \
+        /* op2_live_size */ LIMBSOF(op2) + 2,           \
+        /* res_live_size */ LIMBSOF(ex) - 1,            \
+        /* check_size */ LIMBSOF(ex) - 1,               \
```

Comment:
Looking at the check_limbs_value() doesn't this mean that nothing will be checked as start and end are equal here?

Reply:
Oh way, maybe you mean to say that it's pointless to use `exp`, 'cause that won't be checked anyway.  Oooook, that makes sense!

Reply:
Fixed